rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java

```diff
@@ -159,35 +160,40 @@ public RskSystemProperties rskSystemProperties() {
     }
 
     @Bean
-    public BlockParentDependantValidationRule blockParentDependantValidationRule() {
-        Repository repository = appCtx.getBean(Repository.class);
+    public BlockParentDependantValidationRule blockParentDependantValidationRule(
+            Repository repository,
+            RskSystemProperties config,
+            DifficultyCalculator difficultyCalculator) {
         BlockTxsValidationRule blockTxsValidationRule = new BlockTxsValidationRule(repository);
         PrevMinGasPriceRule prevMinGasPriceRule = new PrevMinGasPriceRule();
         BlockParentNumberRule parentNumberRule = new BlockParentNumberRule();
-        BlockDifficultyRule difficultyRule = new BlockDifficultyRule();
-        BlockParentGasLimitRule parentGasLimitRule = new BlockParentGasLimitRule(RskSystemProperties.CONFIG.getBlockchainConfig().
+        BlockDifficultyRule difficultyRule = new BlockDifficultyRule(difficultyCalculator);
+        BlockParentGasLimitRule parentGasLimitRule = new BlockParentGasLimitRule(config.getBlockchainConfig().
                 getCommonConstants().getGasLimitBoundDivisor());
 
         return new BlockParentCompositeRule(blockTxsValidationRule, prevMinGasPriceRule, parentNumberRule, difficultyRule, parentGasLimitRule);
     }
 
     @Bean(name = "blockValidationRule")
-    public BlockValidationRule blockValidationRule() {
-        BlockStore blockStore = appCtx.getBean(BlockStore.class);
-        int uncleListLimit = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getUncleListLimit();
-        int uncleGenLimit = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getUncleGenerationLimit();
-        int validPeriod = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getNewBlockMaxMinInTheFuture();
+    public BlockValidationRule blockValidationRule(
+            BlockStore blockStore,
+            RskSystemProperties config,
+            DifficultyCalculator difficultyCalculator,
+            ProofOfWorkRule proofOfWorkRule) {
+        int uncleListLimit = config.getBlockchainConfig().getCommonConstants().getUncleListLimit();
```

Comment:
add a var with the value of config.getBlockchainConfig().getCommonConstants() to make it a bit cleaner